float64[] cylinder_data
